### PR TITLE
fix: Using blobs for multipart data

### LINKED ISSUE #12237: [Bug]: Google Drive upload file does not seem to be working
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

With REST APIs now supporting binary data, we expect Google drive uploads to work, but the simple uploads fail to encode data correctly.

### Steps To Reproduce

Generate REST API for this endpoint:
`https://www.googleapis.com/upload/drive/v3/files?uploadType=media`

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by nidhi-nair**
## Description
From [this reference](https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects) for `FormData`, it seems that some sort of conversions are possible when passing values into form data as simply strings. This PR modifies those into blobs to retain native raw format.

We're also handling the proper encoding for data when an unknown content-type is used.

Fixes #12237 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manually tested
- DP available at test.appsmith.com

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/bin-data-as-blobs 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/sagas/ActionExecution/PluginActionSaga.ts | 17.96 **(-0.06)** | 2.31 **(0)** | 9.52 **(0)** | 20.58 **(-0.17)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/sagas/ActionExecution/PluginActionSaga.ts [hunk 1]
```diff
@@ -244,10 +244,13 @@ function* evaluateActionParams(
...
     }
 
-    if (typeof value === "object") value = JSON.stringify(value);
+    if (typeof value === "object") {
+      value = JSON.stringify(value);
+    }
     if (isBlobUrl(value)) {
       value = yield call(readBlob, value);
     }
+    value = new Blob([value], { type: "text/plain" });
```
Comment (AmanAgarwal041) [inline app/client/src/sagas/ActionExecution/PluginActionSaga.ts:253]: If it's a blob url then are we reading and converting it back again to blob object?

**Review by AmanAgarwal041 — APPROVED** [review]

File: app/client/src/sagas/ActionExecution/PluginActionSaga.ts (hunk 1 above)
Comment (nidhi-nair) [reply app/client/src/sagas/ActionExecution/PluginActionSaga.ts:253]: There is a possible optimization that could be done here. I'll create an issue for it and allow someone with more frontend knowledge to take it up separately.

**Comment by nidhi-nair** [comment]
Awaiting QA sign off on DP before merging.

**Comment by nidhi-nair** [comment]
Received sign off on test.appsmith.com by @prapullac

**Pull request merged**